simulation_utils/ParameterParser.h

```diff
@@ -88,8 +83,45 @@ class Simulation_Parameters {
     double controller_dt; // [s]
     double estimator_dt;  // [s]
 
+    // Satellite Parameetr Dispersion distributions
+
+    // Physical
+    std::normal_distribution<double> mass_dist;
+    std::normal_distribution<double> area_dist;
+    std::normal_distribution<double> Ixx_dist;
+    std::normal_distribution<double> Iyy_dist;
+    std::normal_distribution<double> Izz_dist;
+
+    // Actuators
+    std::normal_distribution<double> rw_orientation_dist;
+    std::normal_distribution<double> I_rw_dist;
+    std::normal_distribution<double> mtb_orientation_dist;
+    std::normal_distribution<double> mtb_resistance_dist;
+
+    // Sensors
+    std::normal_distribution<double> gps_pos_dist;
+    std::normal_distribution<double> gps_vel_dist;
+    std::normal_distribution<double> photodiode_orientation_dist;
+    std::normal_distribution<double> photodiode_dist;
+    std::normal_distribution<double> magnetometer_dist;
+    std::normal_distribution<double> gyro_bias_dist;
+    std::normal_distribution<double> gyro_white_noise_dist;
+
+    // Initialization
+    std::normal_distribution<double> sma_dist;
+    std::normal_distribution<double> eccentricity_dist;
+    std::normal_distribution<double> inclination_dist;
+    std::normal_distribution<double> RAAN_dist;
+    std::normal_distribution<double> AOP_dist;
+    std::uniform_real_distribution<double> true_anomaly_dist;
+    std::uniform_real_distribution<double> initial_attitude_dist;
+    std::uniform_real_distribution<double> initial_angular_rate_dist;
+    std::uniform_real_distribution<double> sim_start_time_dist;
+
```

Comment:
just a bit of refactoring, but we can probably find a nicer, systematic way to deal with this where we would define an array of dictionaries in config file, each one with the information on the distribution, ie normal or uniform, reference values, name. It may not be worth the effort but would look nicer and make it easier to add new sources of error to the sim. It's not an essential, I'll add it to the kanban for later. rn not worth losing time with this

Reply:
By refactor, you mean the params.yaml file to have a nested dictionary structure? Or something in the cpp file itself?

Reply:
for the params file to have a sort of nested dictionary structure.